toucan_connectors/soap/soap_connector.py

```diff
@@ -73,11 +74,30 @@ def _retrieve_data(self, data_source: SoapDataSource) -> pd.DataFrame:
         # Force the casting of response
         response = getattr(client.service, data_source.method)(**data_source.parameters)
 
-        if isinstance(response, Sequence):
-            result = pd.DataFrame(response)
-        else:
+        if is_list_response(response):
+            if is_nested_list(response):  # If response is like [['a', 'b',' c', 'd']]
```

Comment:
Is it possible that the response contains not one but multiple nested lists ?

Reply:
Yes, I don't know where to stop regarding unnesting. In `flatten_columns` utils we stayed at 1 level of unnesting.